views/components/Matrix.d3.js

```diff
@@ -1,454 +1,599 @@
-import * as d3 from 'd3';
-import AbstractGraph from './AbstractGraph.d3';
-import { getColor } from '../../server/utils/invertColor';
-import { stringToDate } from '../../server/utils/dateConverter';
-import basePathConfig from '../../server/configs/basePathConfig';
+import * as d3 from 'd3'
 
-const basePath = basePathConfig || '';
+import { getColor } from '../../server/utils/invertColor'
+import { stringToDate } from '../../server/utils/dateConverter'
+import basePathConfig from '../../server/configs/basePathConfig'
+
+const basePath = basePathConfig || ''
 
 const margin = {
   top: 30,
   bottom: 20,
   left: 10,
-  right: 30
+  right: 30,
 }
 
-const cellWidth=65;
+const cellWidth = 65
+const DARK_GREY = '#545454'
+const GREY = '#e6e6e6'
+const NONE = 'none'
+const WHITE = '#ffffff'
+
+export default class Matrix {
+  constructor(el, props) {
+    this.el = el
+    this.cardSize = props.cardSize
+    this.halfCardSize = props.cardSize / 2
+    this.startDay = props.startDay
+    this.lastDay = props.lastDay
+    this.maxDay = props.maxDay
+    this.initialHeight = props.height
+    this.height = props.height - margin.top - margin.bottom
+    this.width = props.width - margin.right
```

Comment:
Do we need to account for left margin?

Reply:
I'm not sure. This should be a pure refactor without any functional changes so this value should be what the existing `width` value is.